lib/src/element.dart

```diff
@@ -14,199 +14,208 @@
-      tree.registerStyle(style);
+      if (newConfiguration.attributes != null && newConfiguration.attributes.isNotEmpty) {
+        newConfiguration.attributes.forEach((String name, String value) {
+          update.updateAttribute(name, value);
+        });
+      }
+
+      _setStyles(newConfiguration, update);
     }
-    nativeNode.classes.add(style.identifierClass);
+    super.update(newConfiguration, update);
   }
 
-  void _updateStyles(Element newConfiguration) {
-    var style = newConfiguration.style;
-    if (!identical(_appliedStyle, style)) {
-      bool hasStyle = _appliedStyle != null;
-      bool willHaveStyle = style != null;
-
-      if (hasStyle && !willHaveStyle) {
-        nativeNode.classes.remove(_appliedStyle.identifierClass);
-      } else if (!hasStyle && willHaveStyle) {
-        _addStyle(style);
-      } else {
-        nativeNode.classes.remove(_appliedStyle.identifierClass);
-        _addStyle(style);
+  void _updateEventListeners(Map<EventType, EventListener> eventListeners, ElementUpdate update) {
+    if (eventListeners != null && eventListeners.isNotEmpty) {
+      if (_baristaId == null) {
+        _baristaId = _nextBid();
+        update.updateBaristaId(_baristaId);
       }
     }
-    _appliedStyle = style;
+  }
 
-    var styles = newConfiguration.styles;
-    if (!identical(_appliedStyles, styles)) {
-      bool hasStyles = _appliedStyles != null && _appliedStyles.isNotEmpty;
-      bool willHaveStyles = styles != null && styles.isNotEmpty;
+  void _updateAttributes(Map<String, String> newAttrs, ElementUpdate update) {
+    final oldAttrs = _configuration.attributes;
+    if (newAttrs != oldAttrs) {
+      // TODO(yjbanov): attribute updates are probaby sub-optimal.
 
-      if (!hasStyles && !willHaveStyles) {
-        return;
+      // Find updates
+      for (String newName in newAttrs.keys) {
+        final newValue = newAttrs[newName];
+        if (oldAttrs[newName] != newValue) {
+          update.updateAttribute(newName, newValue);
+        }
       }
 
-      if (hasStyles && !willHaveStyles) {
-        // Simply remove all
-        for (Style style in _appliedStyles) {
-          nativeNode.classes.remove(style.identifierClass);
-        }
-      } else if (!hasStyles && willHaveStyles) {
-        // Simply add all
-        for (Style style in styles) {
-          _addStyle(style);
+      // Find removes
+      for (final oldName in oldAttrs.keys) {
+        if (!newAttrs.containsKey(oldName)) {
+          // TODO(yjbanov): this won't go far. Need explicit "remove attribute" op.
+          update.updateAttribute(oldName, '');
         }
-      } else {
-        // Do the diffing
-        int i = 0;
-        while (i < _appliedStyles.length && i < styles.length &&
-               identical(_appliedStyles[i], styles[i])) {
-          i++;
+      }
+    }
+  }
+
+  void _updateStyles(Element newConfig, ElementUpdate update) {
+    Style oldStyle = _configuration.style;
```

Comment:
Let's chat about what style we want to go with, Flutter or Effective Dart, clean it up and seal it using the linter. Otherwise, it will be hard to keep it consistent.

https://github.com/yjbanov/butterfly/issues/13